### PR TITLE
SFDP-36 - Validate Inbound Messages

### PR DESCRIPTION
**Pull request by sfitz42**
https://eaflood.atlassian.net/browse/SFDP-36

### CODE REVIEW

**Review by rtasalem — DISMISSED** [review]
LGTM. Only spotted one thing in the `package.json` regarding the Joi version but I think since it was added in I'm not sure if it was needed for any of the validation that was implemented in this PR? Happy to approve either way. If re-approval is needed just let me know. 😊

File: package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "cron": "^3.1.8",
     "ffc-messaging": "2.10.1",
     "hapi-pino": "12.1.0",
-    "joi": "17.13.3",
+    "joi": "^17.13.3",
```
[the remark this answers is not in the document]
Comment (sfitz42) [reply package.json:35]: Good spot!

File: app/schemas/comms-request/v3.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  type: Joi.string().required(),
+  datacontenttype: Joi.string().valid('application/json').required(),
+  time: Joi.string().isoDate().required(),
+  data: Joi.object({
+    crn: crn.optional(),
+    sbi: sbi.required(),
+    sourceSystem: Joi.string().required(),
+    notifyTemplateId: Joi.string().uuid().required(),
+    commsType: Joi.string().valid('email').required(),
+    commsAddresses: Joi.alternatives(
```
[the remark this answers is not in the document]
Comment (sfitz42) [reply app/schemas/comms-request/v3.js:18]: Yep it's pretty much a or in Joi to handle both

File: app/schemas/comms-request/versions.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import v3 from './v3.js'
```
[the remark this answers is not in the document]
Comment (sfitz42) [reply app/schemas/comms-request/versions.js:1]: Just aliased `v3` as `commsSchema` for this as there is just the one at the moment. We will probably change this when we start work on supporting multiple AsyncApi contracts.

**Review by rtasalem — DISMISSED** [review]
LGTM 😊

**Review by rtasalem — DISMISSED** [review]

**Review by rtasalem — APPROVED** [review]
LGTM

**Pull request merged**